engine/src/main/java/org/terasology/engine/module/ExternalApiWhitelist.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.engine.module;
+
+
+import com.google.common.collect.Sets;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.Reader;
+import java.io.StringReader;
+import java.lang.reflect.InvocationTargetException;
+import java.util.Collections;
+import java.util.Set;
+
+public final class ExternalApiWhitelist {
+    public static final Set<Class<?>> CLASSES;
+    public static final Set<String> PACKAGES;
+
+    private ExternalApiWhitelist() {
+    }
+
+    static {
```

Comment:
I would try to avoid static initializers, if possible. A better solution could be `CLASSES = ImmutableSet.builder().add(a).add(b)..`;